### PR TITLE
Spam Comparator rewrite

### PR DESCRIPTION
**Pull request by Msaglier**
## How to test?
Get SmartPhone_xxx_.csv from Stif Alert Trafic.
Add csv files at script roots or anywhere else.
Launch script.
Specified the csv files path.
Specified the date and number of days you want to get.
Specified the file name you want.

At the end, a result file will be generated with:
date;device; number of send messages for this device at this date.

### CODE REVIEW

File: spam_comparator/main.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            return abs(int(days))
+        except ValueError:
+            print('Ce n est pas un nombre valide. \n')
+
+
+
+def request_file_for_result_input():
+    while True:
+        file = input('Quel nom voulez-vous pour le fichier de resultats .txt qui sera deposé à la racine? '
+                     '(Defaut : Results)')
```
Comment (pthegner) [inline spam_comparator/main.py]:
```suggestion
                     '(Defaut : results)')
```

File: spam_comparator/main.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            file = re.search("^[\w,\s-]", file)
+            return file.string
+        except AttributeError:
+            print('Certains caractères ne sont pas valides.\n')
+
+
+def main():
+    print('Bienvenu dans Spam Comparator!\n')
+
+    path = request_path_input()
```
Comment (antoine-de) [inline spam_comparator/main.py]: while this is a completely valid way to have some input args, in general I prefer having classic cli argument as with them it's easier to script.

for this you can check https://github.com/google/python-fire (or https://www.pyinvoke.org/, but in your case, maybe fire is easier)

File: spam_comparator/main.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    date = request_date_input()
+    days = request_days_input()
+    file_name_for_result = request_file_for_result_input()
+
+    spam_comparator = SpamComparator(path, date, days, file_name_for_result)
+    spam_comparator.run()
+
+
+if __name__ == '__main__':
+    DIRECTORY_PATH = './'
```
Comment (antoine-de) [inline spam_comparator/main.py]: iAs this is a constant, t's more pythonic to define this in the beginning of the file

File: spam_comparator/spam_comparator.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    :return: dict with files by wanted date
+    '''
+    files_by_date = {}
+    for file in os.listdir(path):
+        if file.lower().endswith('.csv'):
+            for date in dates:
+                date = date.strftime("%Y%m%d")
+                if date in file:
+                    try:
+                        files_by_date[date].append(file)
```
Comment (antoine-de) [inline spam_comparator/spam_comparator.py]: to make this easier, you can use https://docs.python.org/2/library/collections.html#collections.defaultdict

File: spam_comparator/main.py (hunk 2 above)
Comment (datanel) [reply spam_comparator/main.py]: another one https://click.palletsprojects.com/en/7.x/

**Pull request closed without merging**